main.py

```
@@ -113,53 +114,125 @@ def mine_and_process_block(chain, mempool, miner_pk):
-            peer_host = peer_addr.rsplit(":", 1)[0] if ":" in peer_addr else peer_addr
-            peer_host = peer_host.strip("[]")
-            is_trusted = peer_addr in TRUSTED_PEERS or peer_host in TRUSTED_PEERS
-            is_localhost = peer_host in LOCALHOST_PEERS
-            if chain.state.accounts and not (is_trusted or is_localhost):
-                logger.warning("🔒 Rejected sync from untrusted peer %s", peer_addr)
+        if msg_type == "hello":
+            peer_chain_id = payload.get("chain_id")
+            peer_gen_hash = payload.get("genesis_hash")
+            if peer_chain_id != chain.chain_id:
+                logger.warning("🔒 Disconnecting peer %s: chain_id mismatch (got %s, expected %s)", peer_addr, peer_chain_id, chain.chain_id)
+                asyncio.create_task(network.disconnect_peer(peer_addr))
```

Ruff (0.15.17)

[warning] Store a reference to the return value of `asyncio.create_task`

(RUF006)